x-pack/platform/plugins/shared/lens/public/visualizations/datatable/visualization.tsx

```diff
@@ -126,14 +130,20 @@ export const getDatatableVisualization = ({
 
   triggers: [VIS_EVENT_TO_TRIGGER.filter, VIS_EVENT_TO_TRIGGER.tableRowContextMenuClick],
 
-  initialize(addNewLayer, state) {
-    return (
-      state || {
-        columns: [],
-        layerId: addNewLayer(),
-        layerType: LayerTypes.DATA,
-      }
-    );
+  initialize(addNewLayer, state, mainPalette, datasourceState) {
+    if (state) return convertToRuntimeState(state, datasourceState);
+
+    return {
+      columns: [],
+      layerId: addNewLayer(),
+      layerType: LayerTypes.DATA,
+    };
+  },
+
+  isEqual(state1, references1, datasourceState1, state2, references2, datasourceState2) {
+    const convertedState1 = convertToRuntimeState(state1, datasourceState1);
+    const convertedState2 = convertToRuntimeState(state2, datasourceState2);
```

Comment:
why deep cloning the objects for a comparison? It seems a bit inefficient to me.

Reply:
That's how Marta did it for the xy runtime migration. I'll take a look to see if there is a real need for doing so.

Reply:
The XY clone happens on the inject refs logic.

Removed all `deepClone` calls in 628cb18b339f22e7c51cf60b0935bfa519fdbf19